### ISSUE TITLE
Blob is not defined

### ISSUE THREAD

**Issue by blakeembrey** [aws/aws-sdk-js-v3#1378]
**Describe the bug**

When using the AWS SDK with Cloudflare Workers I get a `Blob is not defined` error.

**SDK version number** `1.0.0-gamma.4`

**Is the issue in the browser/Node.js/ReactNative?** Web worker pseudo environment 

**Details of the browser/Node.js/ReactNative version** N/A

**To Reproduce (observed behavior)** Same as https://github.com/aws/aws-sdk-js-v3/issues/1165#issue-615289280.

**Expected behavior** It should not fail.

**Additional context**

It should be possible to remove `Blob` usage from the main path since it appears to be there primarily for older browser support? https://github.com/aws/aws-sdk-js-v3/issues/1165#issuecomment-657904703.

**Comment by AllanFly120** [comment]
Hi @blakeembrey 

Have you tried [polyfilling]( https://webpack.js.org/guides/shimming/) `Blob` for your runtime? Blob constructor is very wildly supported in browsers, and it's hard to avoid them in the browser code path. Given we don't officially support CloudFlare workers and it's hard to test whether we can do it right, polyfilling the `Blob` is more viable solution.

**Comment by blakeembrey** [comment]
What about in web workers? Also, as far as I can tell, there's only two code paths relying on it and they're both easy to change code paths. One seems to be a handler for older browsers without readable streams, the other was just to count the number of bytes in a string.

**Comment by blakeembrey** [comment]
Also happy to submit a PR with the appropriate changes needed. Polyfilling `Blob` seems like more work than it's worth since I'm not aware of any existing polyfills. Currently I'm just using `aws4fetch` instead, but it's getting tedious having to reimplement each interface (SES, DynamoDB and KMS so far).

**Comment by AllanFly120** [comment]
@blakeembrey 

Sure I'm happy to review the PR. To your question: 
> What about in web workers (I actually have no idea if Blob exists in web workers)? 

I did a little experiment from my console, and I think they are available. Here's what I did:

```console
const response = "self.onmessage=function(e){postMessage('Worker: '+new Blob(['hello world']));}";
let blob;
try {
    blob = new Blob([response], {type: 'application/javascript'});
} catch (e) { // Backwards-compatibility
    window.BlobBuilder = window.BlobBuilder || window.WebKitBlobBuilder || window.MozBlobBuilder;
    blob = new BlobBuilder();
    blob.append(response);
    blob = blob.getBlob();
}
var worker = new Worker(URL.createObjectURL(blob));
worker.onmessage = function(e) {
    alert('Response: ' + e.data);
};
worker.postMessage('Test');
```

**Comment by trivikr** [comment]
@blakeembrey Can you verify if this is fixed in release [v1.0.0-gamma.5](https://github.com/aws/aws-sdk-js-v3/releases/tag/v1.0.0-gamma.5) (i.e. in `@aws-sdk/client-dynamodb@1.0.0-gamma.5`)?

**Comment by blakeembrey** [comment]
I'm not currently using the DynamoDB client, but confirmed it works with KMS and SES now so I bet DynamoDB also works now. Thanks!
